lib/IRGen/GenType.cpp

```diff
@@ -1843,7 +1843,8 @@ TypeCacheEntry TypeConverter::convertAnyNominalType(CanType type,
     case DeclKind::Enum:
       return convertEnumType(type.getPointer(), type, cast<EnumDecl>(decl));
     case DeclKind::Struct:
-      return convertStructType(type.getPointer(), type, cast<StructDecl>(decl));
+      return convertStructType(type.getPointer(), type, cast<StructDecl>(decl),
+                               /*is type dependent*/ decl->isGenericContext());
```

Comment:
This doesn't actually mean its dependent, eg `struct Foo<T> { var x: Int }`

Reply:
That condition is checked further up before we enter this switch. I agree it's best to just do the runtime side of this though.